src/components/views/messages/MVideoBody.tsx

```diff
@@ -124,8 +142,35 @@ export default class MVideoBody extends React.Component {
         }
     }
 
+    async _videoOnPlay() {
+        const autoplay = SettingsStore.getValue("autoplayGifsAndVideos") as boolean;
+        if (autoplay || this.state.decryptedUrl || this.state.fetchingData || this.state.error) {
+            // The video has or will have the data.
+            return;
+        }
+        this.setState({
+            // To stop subsequent download attempts
+            fetchingData: true,
+        });
+        const content = this.props.mxEvent.getContent();
+        if (!content.file) {
+            this.setState({
+                error: "No file given in content",
+            });
```

Comment:
this could be collapsed to just 1 line, but it ultimately is personal preference.